src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -850,6 +879,23 @@ def aks_create(cmd,     # pylint: disable=too-many-locals,too-many-statements,to
     if location is None:
         location = rg_location
 
+    creationData = None
```

Comment:
We are about to discard the old implementation of `aks_create`, please follow this [wiki](https://dev.azure.com/msazure/CloudNativeCompute/_wiki/wikis/CloudNativeCompute.wiki/188955/AZCLI-Coding-Handbook) to write a new implementation using decorator pattern.

Basically, you need to make the following modifications in file `decorator.py`

- Add a new getter function `get_snapshot_id` in class `AKSPreviewContext`.
- Overwrite `get_node_vm_size` in class `AKSPreviewContext`.
- Add a new "set up" function `set_up_snapshot_create_data` in class `AKSPreviewCreateDecorator`.

I would appreciate it if you could also add corresponding test cases for the above modifications in file `tests/latest/test_decorator.py`.

Reply:
@FumingZhang I understand that your new decorator pattern refactoring is just started, but still needs some time to complete. For example, the function **get_node_vm_size** in class AKSPreviewContext that you mentioned doesn't exist in Main branch yet. So I cannot implement new snapshot related commands using new decorator pattern because your re-factoring is still in progress. However the Snapshot Preview is planned for next week so the cli is needed very soon.

Reply:
Discussed offline, will refactor the implementation of this option in another PR.